backend/src/bundles/auth/auth.service.ts

```diff
@@ -1,33 +1,28 @@
 import {
+    type UserFindResponseDto,
     type UserSignInRequestDto,
     type UserSignInResponseDto,
     type UserSignUpRequestDto,
     type UserSignUpResponseDto,
 } from '~/bundles/users/types/types.js';
-import { type UserRepository } from '~/bundles/users/user.repository.js';
 import { type UserService } from '~/bundles/users/user.service.js';
 import { cryptCompare } from '~/common/helpers/helpers.js';
 import { HttpCode, HttpError } from '~/common/http/http.js';
 import { tokenService } from '~/common/services/services.js';
 
 class AuthService {
     private userService: UserService;
-    private userRepository: UserRepository;
 
-    public constructor(
-        userService: UserService,
-        userRepository: UserRepository,
-    ) {
+    public constructor(userService: UserService) {
         this.userService = userService;
-        this.userRepository = userRepository;
     }
 
-    public async signIn(id: number): Promise<UserSignInResponseDto> {
-        const user = await this.userService.findById(id);
-
+    public async signIn(
+        userRequestDto: UserSignInRequestDto,
+    ): Promise<UserSignInResponseDto> {
+        const user = await this.verifyLoginCredentials(userRequestDto);
         return {
-            ...user,
-            token: await tokenService.create({ id }),
+            token: await tokenService.create({ id: user.id }),
```

Comment:
why did you remove the user?

Reply:
https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/100#discussion_r1304781021